arch/x86/kvm/vmx/pkvm_init.c

```diff
@@ -411,6 +415,28 @@ static bool mitigate_bhi(void)
 	return true;
 }
 
+static bool mitigate_eibrs_pbrsb(void)
+{
+	/* Fill RSB after vmexit */
+	set_cpu_cap(&pkvm_sym(boot_cpu_data), X86_FEATURE_RSB_VMEXIT_LITE);
+
+	return true;
+}
+
+static bool mitigate_rfds(void)
+{
+	if (!(x86_read_arch_cap_msr() & ARCH_CAP_RFDS_CLEAR))
+		return false;
+
+	set_cpu_cap(&pkvm_sym(boot_cpu_data), X86_FEATURE_CLEAR_CPU_BUF);
+	return true;
+}
+
+static bool mitigate_vmscape(void)
+{
+	return mitigate_spectre_v2_user();
```

Comment:
probably nit: this means that mitigate_spectre_v2_user() we be called twice? In particular, `switch_vcpu_ibpb` will be patched twice?

That is probably harmless less, but still, we could probably avoid that by letting mitigate_spectre_v2_user() clear X86_BUG_VMSCAPE (and print "mitigated vmscape when mitigating spectre_v2_user"), like we do in some other cases?

Reply:
Yeah, makes sense. Probably may also need to let mitigate_vmscape() to clear X86_BUG_SPECTRE_V2_USER if the mitigate_vmscape() is called first, although it is called after spectre_v2_user but seems this is not a strong guarantee and can be changed if we change the order in the possible_cpu_bugs[].